crates/rome_js_analyze/src/analyzers/nursery/no_redundant_use_strict.rs

```diff
@@ -55,44 +73,45 @@ declare_rule! {
     }
 }
 
-declare_node_union! { AnyNodeWithDirectives = JsFunctionBody | JsModule | JsScript }
+declare_node_union! { AnyNodeWithDirectives = JsFunctionBody | JsScript }
 impl AnyNodeWithDirectives {
     fn directives(&self) -> JsDirectiveList {
         match self {
             AnyNodeWithDirectives::JsFunctionBody(node) => node.directives(),
             AnyNodeWithDirectives::JsScript(script) => script.directives(),
-            AnyNodeWithDirectives::JsModule(module) => module.directives(),
         }
     }
 }
+declare_node_union! { JsModuleOrClass = JsClassDeclaration | JsClassExpression| JsModule  }
```

Comment:
```suggestion
declare_node_union! { AnyJsStrictModeNode = AnyJsClass| JsModule | JsDirective }
```

You can add `JsDirective` so that you can use `AnyJsStrictModeNode` as the `State` instead of the generic `SyntaxNode`